gradle-sls-packaging/build.gradle

```diff
@@ -23,6 +26,16 @@ dependencies {
     testCompile 'org.awaitility:awaitility'
 }
 
+
+/**
+ * Per default, Gradle TestKit excludes compileOnly dependencies resulting in ClassNotFoundExceptions when sls-packaging
+ * tries to access 'gradle-consistent-versions' classes in integ tests e.g. in 'AssetDistributionPluginIntegrationSpec'.
+ * https://discuss.gradle.org/t/testkit-somehow-excludes-compileonly-dependencies
+ */
+tasks.withType(PluginUnderTestMetadata).configureEach {
+    pluginClasspath.from(configurations.testRuntime)
```

Comment:
This is pretty hacky but I could not get it to work otherwise.

The problem is that TestKit does not put compileOnly dependencies on the same classpath as the plugin under test which results in a `NoClassDefFoundError: com/palantir/gradle/versions/VersionsLockExtension` when running tests ([circle](https://app.circleci.com/pipelines/github/palantir/sls-packaging/743/workflows/b23dece6-1c41-426c-bc4f-021501baa28a/jobs/10049)).

I tried adding the GCV plugin via buildscript dependency or just as a `testCompile/testRuntime` dependency but got the same error. Using `pluginClasspath.from(configurations.compileOnly)` also fails at configuration time with `Could not resolve all files for configuration ':gradle-sls-packaging:compileOnly'. > Could not find com.palantir.sls-packaging:sls-packaging:4.16.2-20-g3611d3b.dirty`.

Not that this is only an issue in tests. When ran on a real project, gcv classes are on the same classpath when `project.getRootProject().getPlugins().withId("com.palantir.consistent-versions", plugin -> { //... })` gets evaluated.